app/src/main/java/at/team30/setroute/infrastructure/ImageRepository.kt

```diff
@@ -20,29 +22,28 @@ class ImageRepository @Inject constructor( private val routesRepository : IRoute
         val route = routesRepository.getRoutesById(route_id)
         if(route != null && image == null)
         {
-            fun run() {
-                var url = "https://maps.googleapis.com/maps/api/staticmap?size=600x400&key=" + apiKey + "&style=feature:all%7Cvisibility=off"
-                var counter = 0
-                if (route.positions != null) {
-                    route.positions.forEach {
-                        url = url + "&markers=color:red%7Clabel:" + (++counter) + "%7C" + it.latitude + "," + it.longitude
-                    }
+            var url = "https://maps.googleapis.com/maps/api/staticmap?size=600x400&key=" + apiKey + "&style=feature:all%7Cvisibility=off"
+            var counter = 0
+            if (route.positions != null) {
+                route.positions.forEach {
+                    url = url + "&markers=color:red%7Clabel:" + (++counter) + "%7C" + it.latitude + "," + it.longitude
                 }
+            }
+
+
+
+            val request = Request.Builder().url(url).build()
 
-                val request = Request.Builder()
-                    .url(url)
-                        .build()
-
-                            client.newCall(request).execute().use { response ->
-                        if (!response.isSuccessful) throw IOException("Unexpected code $response")
-                                /*
-                        for ((name, value) in response.headers) {
-                            println("$name: $value")
-                        }
-                                 */
-                         image = BitmapFactory.decodeStream(response.body!!.byteStream())
-                    }
+            var response = client.newCall(request).execute()
+
+            if (!response.isSuccessful) throw IOException("Unexpected code $response")
```

Comment:
I think this Exception is never called so probably don't throw it. We have a fallback-image anyways

just return null if possible :)

Reply:
Nevermind, just saw that the exception is caught. My bad :)